pageserver/src/tenant/timeline.rs

```diff
@@ -1709,10 +1709,7 @@ impl Timeline {
         guard.initialize_local_layers(loaded_layers, disk_consistent_lsn + 1);
 
         if let Some(rtc) = self.remote_client.as_ref() {
-            rtc.schedule_layer_file_deletion(&needs_cleanup)?;
-            rtc.schedule_index_upload_for_file_changes()?;
-            // Tenant::create_timeline will wait for these uploads to happen before returning, or
-            // on retry.
+            rtc.flushing_delete_layers(&needs_cleanup).await?;
```

Comment:
The delta between `main` and proposed is that we would not delete these from the index_part.json.

Also, this would slow down startup.

Reply:
I wonder if there's an issue with deletion queue being persistent and this not necessarily being or perhaps this is why you omitted the index part scheduling... If we omitted it, when starting from the same version we would always want to delete the same future layers. Should we advance in remote consistent lsn, we would upload a L0 and the index_part.json (including this earlier deletion), so this would work.

Reply:
Also if there should be a gc or compaction update this should work because they upload an index_part.json. I cannot see other ways.